### PR TITLE
Allow `EnemyUIName` to be used for other TechnoTypes

### LINKED ISSUE #503: [New Feature] UIName that only visible to the enemy
### Description:
In Ares, we can use EnemyUIName=, non-friendly parties display this name when they look at the building, not the real name. But this flag only can used on buildingtypes. Maybe it will be more usefull if we got a new flag that can used on all technotypes. Just like in Tiberian Sun.

### Possible Implementation:

```ini
[SomeTechnoType]
EnemyUIName=
(or other flag)
```

### PR DESCRIPTION
**Pull request by Otamaa**
close #503
`EnemyUIName`  not applied if  `Disguised` , `Civilian/MultiplayPassive Owned` , `Observer` !

```INI
Rules
[UNIT/INFANTRY/AIRCRAFT TYPE]
EnemyUIName= (csf string)
```

### CODE REVIEW

**Comment by CommanderNour** [comment]
SCRN 20220117-084111-00473

Works

File: src/Ext/TechnoType/Body.h [hunk 1]
```diff
@@ -163,6 +164,7 @@ class TechnoTypeExt
 			, DeployingAnim_ReverseForUndeploy { true }
 			, DeployingAnim_UseUnitDrawer { true }
 			, DeployDir {}
+			, EnemyUIName{ }
```
Comment (Starkku) [inline src/Ext/TechnoType/Body.h:167]: Move space from between the curly braces to before.

File: src/Ext/TechnoType/Hooks.cpp [hunk 2]
```diff
@@ -302,4 +302,42 @@ DEFINE_HOOK(0x739C86, UnitClass_DeployUndeploy_DeploySound, 0x6)
 		return 0; // Only play sound when done with deploying or undeploying.
 
 	return isDeploying ? DeployReturn : UndeployReturn;
+}
+
+// Issue #503
+// Author : Otamaa
+DEFINE_HOOK(0x4AE670, DisplayClass_GetToolTip_EnemyUIName, 0x8)
+{
+	enum {SetUIName = 0x4AE678 };
```
Comment (Starkku) [inline src/Ext/TechnoType/Hooks.cpp:311]: Add space before `SetUIName`.

File: src/Ext/TechnoType/Hooks.cpp [hunk 3]
```diff
@@ -302,4 +302,42 @@ DEFINE_HOOK(0x739C86, UnitClass_DeployUndeploy_DeploySound, 0x6)
...
+		if (auto pOwnerHouse = pFoot->GetOwningHouse())
+		{
+			IsAlly = pOwnerHouse->IsAlliedWith(HouseClass::Player);
+			IsCivilian = (pOwnerHouse == HouseClass::FindCivilianSide()) || pOwnerHouse->IsNeutral();
+		}
+
+		if (!IsAlly && !IsCivilian && !IsObserver)
+		{
+			auto pTechnoTypeExt = TechnoTypeExt::ExtMap.Find(pTechnoType);
+			if (auto pEnemyUIName = pTechnoTypeExt->EnemyUIName.Get().Text)
```
Comment (Starkku) [inline src/Ext/TechnoType/Hooks.cpp:334]: Add empty line between the variable declaration and if clause.

**Review by Starkku — CHANGES_REQUESTED** [review]
Code looks okay, just some minor code style tweaks required.

**Review by Starkku — APPROVED** [review]
Approving changes & merging.

**Pull request merged**